relay-crash/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+//! Native crash reporting for Relay.
+//!
+//! Use [`CrashHandler`] to configure and install a crash handler.
+
+use std::path::Path;
+
+#[cfg(unix)]
+mod native {
+    // Code generated by bindgen contains some warnings and also offends the linter. Ignore all of
+    // that since they do not have a consequence on the functions used for relay-crash.
+    #![allow(warnings, clippy::all)]
+
+    include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+}
+
+/// Captures process crashes and reports them to Sentry.
+///
+/// Internally, this uses the Breakpad client to capture crash signals and send minidumps to Sentry.
+/// If no DSN is configured, the crash handler is not initialized.
+#[derive(Debug, Default)]
+pub struct CrashHandler<'a> {
+    dsn: &'a str,
+    database: &'a str,
+    release: Option<&'a str>,
+    environment: Option<&'a str>,
+}
+
+impl<'a> CrashHandler<'a> {
+    /// Creates a new crash handler.
+    ///
+    /// Panics if there is are non UTF-8 characters in the path.
+    pub fn new(dsn: &'a str, database: &'a Path) -> Self {
+        Self {
+            dsn,
+            database: database.to_str().unwrap(),
+            release: None,
+            environment: None,
+        }
+    }
+
+    /// Set the crash handler's Sentry release.
+    ///
+    /// Defaults to no release.
+    pub fn release(&mut self, release: Option<&'a str>) -> &mut Self {
+        self.release = release;
+        self
+    }
+
+    /// Set the crash handler's Sentry environment.
+    ///
+    /// Defaults to no environment
+    pub fn environment(&mut self, environment: Option<&'a str>) -> &mut Self {
+        self.environment = environment;
+        self
+    }
+
+    /// Installs the crash handler in the process if a Sentry DSN is set.
+    #[cfg(unix)]
+    pub fn install(&self) {
+        use std::ffi::CString;
+
+        unsafe {
+            let options = native::sentry_options_new();
+
+            let dsn_cstr = CString::new(self.dsn).unwrap();
+            native::sentry_options_set_dsn(options, dsn_cstr.as_ptr());
+
+            let db_cstr = CString::new(self.database).unwrap();
+            native::sentry_options_set_database_path(options, db_cstr.as_ptr());
+
+            if let Some(release) = self.release {
+                let release_cstr = CString::new(release).unwrap();
+                native::sentry_options_set_release(options, release_cstr.as_ptr());
+            }
+
+            if let Some(environment) = self.environment {
+                let env_cstr = CString::new(environment).unwrap();
+                native::sentry_options_set_environment(options, env_cstr.as_ptr());
+            }
+
+            native::sentry_init(options);
+        }
+    }
+
+    #[cfg(not(unix))]
+    pub fn install(&self) {
+        unimplemented!("crash handler not supported on this platform");
```

Comment:
is it even possible to compile this module on windows?

Reply:
it's possible, sentry-native with breakpad works on Windows, too. Given this is fully internal, I just did not build this out in the build script (requires linking against a few other libraries).